opm/autodiff/BlackoilPropsAdFromDeck.cpp

```diff
@@ -64,7 +65,49 @@ namespace Opm
                                                      const Dune::CpGrid& grid,
                                                      const bool init_rock )
     {
-        init(deck, eclState, grid.numCells(), static_cast<const int*>(&grid.globalCell()[0]),
+        auto materialLawManager = std::make_shared<MaterialLawManager>();
+        unsigned number_of_cells = grid.size(0);
+        std::vector<int> compressedToCartesianIdx(number_of_cells);
+        for (unsigned cellIdx = 0; cellIdx < number_of_cells; ++cellIdx) {
+            compressedToCartesianIdx[cellIdx] = grid.globalCell()[cellIdx];
+        }
+        materialLawManager->initFromDeck(deck, eclState, compressedToCartesianIdx);
+        init(deck, eclState, materialLawManager, grid.numCells(), static_cast<const int*>(&grid.globalCell()[0]),
+             static_cast<const int*>(&grid.logicalCartesianSize()[0]),
```

Comment:
I do not remember any more what these `static_cast`s are needed for. Seems a simple call to data() would be ok?